src/language-js/print/property.js

```diff
@@ -94,17 +104,125 @@ function printPropertyKey(path, options, print) {
   return print("key");
 }
 
+// [prettierx] for alignObjectProperties option
+// with computedPropertySpacing option support
+function getPropertyPadding(options, path) {
+  if (!options.alignObjectProperties) {
+    return "";
+  }
+
+  if (/json/.test(options.parser)) {
+    return "";
+  }
+
+  const node = path.getValue();
+  const { type } = node;
+
+  // grandparent node:
+  const parentObject = path.getParentNode(1);
+
+  const { locStart, locEnd } = options;
+
+  // THIS IS A HACK:
+  const shouldBreak = options.originalText
+    .slice(locStart(parentObject), locEnd(parentObject))
+    .match(/{\s*(\/.*)?\n/);
+
+  if (!shouldBreak) {
+    return "";
+  }
+
+  const nameLength =
+    type === "Identifier"
+      ? node.name.length
+      : node.raw
+      ? node.raw.length
+      : node.extra.raw
+      ? node.extra.raw.length
+      : undefined;
+
+  // [prettierx] computedPropertySpacing option support
+  const computedPropertyOverhead = options.computedPropertySpacing ? 4 : 2;
+
+  // FUTURE TBD from arijs/prettier-miscellaneous#10
+  // (does not seem to be needed to pass the tests):
+  // if (nameLength === undefined) {
+  //   return "";
+  // }
```

Comment:
I would like to make an update to throw an explicit error in case this assumption fails.